src/Nest/Aggregations/AggregationContainer.cs

```diff
@@ -287,6 +290,29 @@ public void Accept(IAggregationVisitor visitor)
 		{
 			if (visitor.Scope == AggregationVisitorScope.Unknown) visitor.Scope = AggregationVisitorScope.Aggregation;
 			new AggregationWalker().Walk(this, visitor);
+		}
+
+		internal static ConcurrentDictionary<Type, Type> _aggregateNameCache = new ConcurrentDictionary<Type, Type>();
+		internal static void InjectMetadata(IAggregation aggregation)
+		{
+			aggregation.Meta = aggregation.Meta ?? new Dictionary<string, object>();
+			var typeKey = "_type";
+			if (aggregation.Meta.ContainsKey(typeKey))
+				throw new ArgumentException($"Metadata key: {typeKey} is reserved. Please choose another.");
+			var aggregationType = aggregation.GetType();
+			Type aggregateType;
+			if (!_aggregateNameCache.TryGetValue(aggregationType, out aggregateType))
+			{
+				var attribute = aggregationType.GetTypeInfo().GetInterfaces()
+					.Where(t => typeof(IAggregation).IsAssignableFrom(t))
+					.SelectMany(t => t.GetCustomAttributes(typeof(AggregateTypeAttribute)))
+					.FirstOrDefault() as AggregateTypeAttribute;
+				if (attribute == null)
+					throw new Exception($"{aggregationType} is missing the required AggregateType attribute.");
+				aggregateType = attribute.Type;
+				_aggregateNameCache.TryAdd(aggregationType, aggregateType);
+			}
+			aggregation.Meta.Add(typeKey, aggregateType.FullName);

```

Comment:
I still think it would be nicer to see:

```
"_type" : "extended_stats"
```

in the output json then 

```
"_type" : "Nest.ExtendedStatsAggration"
```

or 

```
"_type" : "Nest.ValueAggregate"
```

I think the first can proof more useful to a user as well.